### ISSUE TITLE
DOTA 2 Keeps Crashing In Game (Mac OS)

### ISSUE THREAD

**Issue by Vuccappella** [ValveSoftware/Dota-2#1691]
Ok so every other game the dota2 Client for mac crashesh in game on top of that almost every game after it has finished it also crashes but no crash log is produced in the later case. The only crash log I have is from when it crashes in the actual game.

https://gist.github.com/Vuccappella/b01929cc558b9beca748 - Crash log.

And my system information:

Processor Information:
    Model:  iMac13,2
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x3a
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  3200 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    
Network Information:
    Network Speed:  
    
Operating System Version:
    MacOS 10.10.1 (64 bit)
    
Video Card:
    Driver:  NVIDIA GeForce GTX 680MX

    Driver Version:  1.2
    Driver Date: 0  0
    OpenGL Version: 2.1
    Desktop Color Depth: 32 bits per pixel
    Monitor Refresh Rate: 0 Hz
    VendorID:  0x10de
    DeviceID:  0x11a3
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  2560 x 1440
    Desktop Resolution: 2560 x 1440
    Primary Display Size: 23.46" x 13.19"  (26.89" diag)
                                            59.6cm x 33.5cm  (68.3cm diag)
    Primary VRAM: 2047 MB
    
Memory:
    RAM:  24576 Mb
    
Miscellaneous:
    UI Language:  English
    Microphone:  Not set
    Media Type:  DVD
    Total Hard Disk Space Available:  2860668 Mb
    Largest Free Hard Disk Block:  1427878 Mb

**Comment by SsnL** [comment]
Crash: https://gist.github.com/SsnL/3d632e8898620fd3be0f

**Comment by Vuccappella** [comment]
After the last few mini-updated the issues have gotten worse and most of the time it doesn't produce crash logs. A common time for it to crash is just after the picking phase ends and you select your hero, the game just crashes. Anyone experiencing that?  It crashes twice per game now, really unplayable.

**Comment by alkore** [comment]
Yes I'm experiencing the exact same issues Vuccappella, usually it's after picking heroes or sometimes random only once during game and only after a 2nd game in hero. Usually first game is fine all the way through and 2nd game will either crash after hero pick or once in game. Then it's OK until the next one.

**Comment by evrial** [comment]
I experience random crashes on Mac OS client since November I guess, maybe earlier. Nothing have changed in that direction by Valve to improve stability neither random crashes caused by memory management.

Just one of hundred crash logs https://gist.github.com/evrial/60aeb14f85d041827a8d

**Comment by alkore** [comment]
yep since november here too.

**Comment by Vuccappella** [comment]
Yes, same for me as well. I just hope these issues are reviewed.

**Comment by Vuccappella** [comment]
Is ther

**Comment by Vuccappella** [comment]
Is there any news about this issue. Dota 2 Is unplayable. The crash logs I've provided are just maybe half of the actual crashes I got since the 18.02.

**Comment by Vuccappella** [comment]
Can we get some info if this is being reviewed ? Or at least a temporary solution to the problem or any way I could possibly help apart from crash logs, anything.

**Comment by krikaoli** [comment]
I spent a lot of money at Dota 2, and it is not nice see how negligent Valve is being with Mac users... I can't play more than two games without a crash... a lot of different crashes. I play using Yosemite and my two most common crashes are: Exception Type: EXC_BAD_ACCESS (SIGSEGV) and EXC_CRASH (SIGABRT)

Process: dota_osx [1255]
Path: /Users/USER/Library/Application Support/Steam/*/dota_osx
Identifier: dota_osx
Version: ???
Code Type: X86 (Native)
Parent Process: bash [1252]
Responsible: steam_osx [931]
User ID: 501
Date/Time: 2015-03-13 04:58:57.335 -0300
OS Version:  Mac OS X 10.10.3 (14D87p)
Report Version: 11
Anonymous UUID:  1BE5E688-0521-B172-BA21-7115C7505D4D
Time Awake Since Boot: 23000 seconds
Crashed Thread: 17

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

**Comment by krikaoli** [comment]
The only solution I found so far it is downgrading Yosemite to Maverick... but not viable to me...

**Comment by Vuccappella** [comment]
Yeah there's actually no way I'm going to downgrade just for dota.

**Comment by krikaoli** [comment]
I have been dealing with the Apple Support to solve this puzzle.

Try to complete remove flash, that could work, but you loose all window info inside the game windows.

Enviado do meu iPhone

**Comment by Vuccappella** [comment]
Sadly I can't do that either, I work with flash from time to time and I need it from time to time as well. Has it solved the issue for you ?

**Comment by krikaoli** [comment]
Not a single crash. I have other idea. Just create a new user with admin power and do that.

Enviado do meu iPhone

**Comment by Vuccappella** [comment]
What do you mean by that? Create a new user, disable flash on that user and then play the game ?

**Comment by krikaoli** [comment]
At settings you have the Users and Groups option, just create another user and install Dota for it.

Enviado do meu iPhone

**Comment by krikaoli** [comment]
But at this new user, you can't install Flash player from Adobe. Got it?

Enviado do meu iPhone

**Comment by Vuccappella** [comment]
I see. I will try it.

**Comment by gxgani** [comment]
Earth Spirit Bug still exists! Ty volvo!

**Comment by cmbirk** [comment]
+1 on the crashing thread.

**Comment by bklooster** [comment]
When you see in the crash log that RAM at 3.7GB it usually means that the 32-bit DOTA process ran out of memory. This is similar to issue #976 and was commented at the time that they are working to try to reduce RAM usages of the app. 

The only workaround is to lower graphical settings (textures/resolution being the bigger ones) to try to reduce the RAM requirements. Don't browse the store/inventory or any other webview or steam overlay that will cause RAM usage to increase before play. 

The devs have been unresponsive lately. We know that a good chunk of the DOTA 2 team is working on Source2, but we don't know if that will include a new updated OpenGL backend for OS X or not, or if the .exe will be 64-bit (thus allowing you to use all of your RAM rather than limited to the current amount). Valve's OS X support/communication has been lacking since about 6mo after steam released for OSX. Probably due to Apple's continued lockdown/walled garden approach to their OSes not jiving with Valve's corporate perspective. In addition Apple has been showing a lot of neglect to OS X OpenGL support and drivers.

Best case scenario for the future, is that source 2 will have a new 64-bit OS X backend. Also that Apple will announce OS X Vulkan support at WWDC2015 with Valve supporting it on release.

**Comment by ESToFI** [comment]
The most reliable (but sadly not foolproof) method to avoid the crash is to restart Dota between games.  Recently, I have been using the following shell script to warn when the dota_osx process is using close to the 3.7G crash-point so I can at least ask for a pause and restart, rather than crashing halfway through a gank or teamfight - maybe it will be useful for others having this issue.

It checks the dota_osx process every 5 seconds for dangerously high memory usage; if the memory usage is over 3.6G (which from a little testing tends to happen shortly before the crash), it will then warn you by speaking "Memory Low" every two minutes.  It doesn't hook into the process or interact with it in any way, just monitors total memory usage by scraping the output of 'ps', so you can just set the script running in a minimised terminal window and forget about it.  If it can't find the process it will wait two minutes and look again, so you can freely restart the game and it won't affect the script.

    #!/bin/sh
    
    echo "Monitoring dota memory"
    while true; do
     DOTAMEM=`ps x -ovsz,comm | grep dota_osx | sed 's/ *\([0-9]*\) *.*/\1/'`
     if (( DOTAMEM > 0 )); then
      if (( DOTAMEM > 3600000)); then
       echo "Warning - Dota memory now $DOTAMEM"
       say "Memory Low"
       sleep 120
      fi
     else
      echo "Can't find dota_osx process - sleeping for 120s"
      sleep 120
     fi
     sleep 5
    done

**Comment by krikaoli** [comment]
I don't have enough words. Thank you very much for your solution and kind reply.
My best regards friend ________________________ Cristiano Caiado De AcioliAdvogado Associado - OAB/DF 31497 (061) 9976-6474 Brasília - DF 


     Em Quinta-feira, 11 de Junho de 2015 11:41, ESToFI <notifications@github.com> escreveu:
   

 The most reliable (but sadly not foolproof) method to avoid the crash is to restart Dota between games. Recently, I have been using the following shell script to warn when the dota_osx process is using close to the 3.7G crash-point so I can at least ask for a pause and restart, rather than crashing halfway through a gank or teamfight - maybe it will be useful for others having this issue.It checks the dota_osx process every 5 seconds for dangerously high memory usage; if the memory usage is over 3.6G (which from a little testing tends to happen shortly before the crash), it will then warn you by speaking "Memory Low" every two minutes. It doesn't hook into the process or interact with it in any way, just monitors total memory usage by scraping the output of 'ps', so you can just set the script running in a minimised terminal window and forget about it. If it can't find the process it will wait two minutes and look again, so you can freely restart the game and it won't affect the script.#!/bin/sh

echo "Monitoring dota memory"
while true; do
 DOTAMEM=`ps x -ovsz,comm | grep dota_osx | sed 's/ *\([0-9]*\) *.*/\1/'`
 if (( DOTAMEM > 0 )); then
  if (( DOTAMEM > 3600000)); then
   echo "Warning - Dota memory now $DOTAMEM"
   say "Memory Low"
   sleep 120
  fi
 else
  echo "Can't find dota_osx process - sleeping for 120s"
  sleep 120
 fi
 sleep 5
done
